src/layers/base-layer.ts

```diff
@@ -182,8 +180,7 @@ class Layer<LayerConfig extends LayerBaseConfig = LayerBaseConfig> {
   id: string;
   // TODO: define meta
   meta: {};
-  // TODO: define visConfigSettings
-  visConfigSettings: {};
+  visConfigSettings: Partial<LayerVisConfigSettings>;
```

Comment:
Each layer has different propertues in visConfig and visConfigSettings. We should make the type here more generic and add specific types for individual layer's `layer.visConfigSettings`

```js
visConfigSettings: Record<string, VisConfig>;
```

Reply:
then in ArcLayer:

```
type ArcLayerVisConfigSettings = {
  opacity: VisConfigNumber;
  thickness: VisConfigNumber;
  colorRange: VisConfigColorRange;
  sizeRange: VisConfigRange;
  targetColor: VisConfigColorSelect;
};

export default class ArcLayer extends Layer {
  visConfigSettings: ArcLayerVisConfigSettings;
...
}
```

Reply:
I have considered your suggestions. This is how I see the implementation of ArcLayer

`type ArcLayerVisConfigSettings = {
  opacity: VisConfigNumber;
  thickness: VisConfigNumber;
  colorRange: VisConfigColorRange;
  sizeRange: VisConfigRange;
  targetColor: VisConfigColorSelect;
};

export default class ArcLayer extends Layer {
  declare visConfigSettings: ArcLayerVisConfigSettings;
...
}`